src/Lykke.Job.TransactionHandler/Sagas/EthereumCoreSaga.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Common;
+using Common.Log;
+using JetBrains.Annotations;
+using Lykke.Common.Chaos;
+using Lykke.Cqrs;
+using Lykke.Job.TransactionHandler.Commands;
+using Lykke.Job.TransactionHandler.Commands.EthereumCore;
+using Lykke.Job.TransactionHandler.Core.Domain.Ethereum;
+using Lykke.Job.TransactionHandler.Core.Domain.PaymentSystems;
+using Lykke.Job.TransactionHandler.Core.Services.BitCoin;
+using Lykke.Job.TransactionHandler.Events;
+using Lykke.Job.TransactionHandler.Events.EthereumCore;
+using Lykke.Service.Assets.Client;
+
+namespace Lykke.Job.TransactionHandler.Sagas
+{
+    public class EthereumCoreSaga
+    {
+        private readonly ILog _log;
+        private readonly IEthereumCashinAggregateRepository _ethereumCashinAggregateRepository;
+
+        public EthereumCoreSaga(
+            [NotNull] ILog log,
+            [NotNull] IEthereumCashinAggregateRepository ethereumCashinAggregateRepository)
+        {
+            _log = log?.CreateComponentScope(nameof(EthereumCoreSaga)) ?? throw new ArgumentNullException(nameof(log));
+            _ethereumCashinAggregateRepository = ethereumCashinAggregateRepository;
+        }
+
+        private async Task Handle(CashinDetectedEvent evt, ICommandSender sender)
+        {
+            Stopwatch sw = new Stopwatch();
+            try
+            {
+                sw.Start();
+
+                _log.WriteInfo(evt.TransactionHash, evt, "Eth Cashin startetd");
```

Comment:
The same thing about log entry process parameter as above. It belongs to the all log writes in this class

Reply:
fixed